File: src/main/kotlin/com/example/menuservice/service/ProductService.java
package com.example.menuservice.service;


import com.example.menuservice.model.Product;
import com.example.menuservice.repo.ProductRepo;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService {

    private final ProductRepo productRepository;

    public ProductService(ProductRepo productRepository) {
        this.productRepository = productRepository;
    }

    //post product
    public Product saveProduct(Product product) {
        return productRepository.save(product);
    }

    //get products
    public List<Product> getProducts() {
        return productRepository.findAll();
    }

    //get product by id
    public Product getProductById(int id) {
        return productRepository.findById(id).orElse(null);
    }

    //find products by category id
    public List<Product> findByCategoryId(int id) {
        return productRepository.findByCategoryId(id);
    }

    //delete product
    public String deleteProduct(int id) {
        productRepository.deleteById(id);
        return "Product deleted" + id;
    }

    //update user
    public Product updateProduct(Product product) {
        Product existingProduct=productRepository.findById(product.getId()).orElse(null);
        existingProduct.setName(product.getName());
        existingProduct.setDescription(product.getDescription());
        existingProduct.setAllergies(product.getAllergies());
        existingProduct.setNutrition(product.getNutrition());
        existingProduct.setPrice(product.getPrice());
        existingProduct.setImage(product.getImage());
        existingProduct.setInStock(product.getInStock());
        return productRepository.save(existingProduct);
    }
}

File: src/main/kotlin/com/example/menuservice/service/ProductService.kt
package com.example.menuservice.service

import com.example.menuservice.model.Product
import com.example.menuservice.repo.ProductRepo
import org.springframework.stereotype.Service

@Service
class ProductService(private val productRepository: ProductRepo) {
    //post product
    fun saveProduct(product: Product): Product {
        return productRepository.save(product)
    }

    //get products
    val products: List<Product>
        get() = productRepository.findAll()

    //get product by id
    fun getProductById(id: Int): Product {
        return productRepository.findById(id).orElse(null)
    }

    //find products by category id
    fun findByCategoryId(id: Int): List<Product> {
        return productRepository.findByCategoryId(id)
    }

    //delete product
    fun deleteProduct(id: Int): String {
        productRepository.deleteById(id)
        return "Product deleted$id"
    }

    //update user
    fun updateProduct(product: Product): Product {
        val existingProduct = productRepository.findById(product.id!!).orElse(null)
        existingProduct.name = product.name
        existingProduct.description = product.description
        existingProduct.allergies = product.allergies
        existingProduct.nutrition = product.nutrition
        existingProduct.price = product.price
        existingProduct.image = product.image
        existingProduct.inStock = product.inStock
        existingProduct.category = product.category
        
        return productRepository.save(existingProduct)
    }
}